### PR TITLE
INTL-78: Exclude some bad cases from intl_message_migration

### PR DESCRIPTION
**Pull request by alanknight-wk**
## Motivation
  
Among the cases we've run into in usage:
 - SelectOption.value should not be translated, although value is usually a translatable attribute
 - FormComponentDisplayPropsMixin.label is sometimes translated, unless hideLabel is true, in which case it isn't, and seems to be used as a program idenitifer.
 - If a string is a URL, even if it's in a translatable attribute and will be presented to the user, we shouldn't try to translate it.

## Changes
  
For the label case we translate it, but add a FIXME for a manual check. For the other two cases we detect them and do not translate.
There are a lot of format changes in this, I don't know why.

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Client Platform member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/over_react_codemod/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react_codemod/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: lib/src/intl_suggestors/utils.dart
Comment (alanknight-wk) [inline lib/src/intl_suggestors/utils.dart:287]: This is a real change.

File: lib/src/intl_suggestors/utils.dart
Comment (alanknight-wk) [inline lib/src/intl_suggestors/utils.dart:6]: I've flagged the real changes in here, to distinguish from the formatting ones.

File: lib/src/intl_suggestors/utils.dart
Comment (alanknight-wk) [inline lib/src/intl_suggestors/utils.dart:27]: This is real - exclude URLS.

File: lib/src/intl_suggestors/utils.dart
Comment (alanknight-wk) [inline lib/src/intl_suggestors/utils.dart:39]: This is real.

File: lib/src/intl_suggestors/utils.dart
Comment (alanknight-wk) [inline lib/src/intl_suggestors/utils.dart:47]: This is real.

File: lib/src/util.dart
Comment (alanknight-wk) [inline lib/src/util.dart:438]: Not sure this is the best way to check this, but it seems to work, and be useful. I even wrote a test for it, since it's going into the more general codemod infrastructure.

File: test/util_test.dart
Comment (alanknight-wk) [inline test/util_test.dart:41]: It's hard to set up a resolved AST on a string in memory, so I made a separate file.

File: lib/src/util.dart
Comment (greglittlefield-wf) [reply lib/src/util.dart:438]: We have this existing extension which does something very similar: https://github.com/Workiva/over_react_codemod/blob/master/lib/src/util/element_type_helpers.dart#L26

so it might make sense to consolidate them.

File: test/util_test.dart
Comment (greglittlefield-wf) [reply test/util_test.dart:40]: FYI there's `SharedAnalysisContext` which makes it a bit easier. 
Example: 
- Setup: https://github.com/Workiva/over_react_codemod/blob/5e30958979a650d0850fadabb068f6ea17d542a7/test/util/element_type_helpers_test.dart#L57-L67
- Usage: https://github.com/Workiva/over_react_codemod/blob/5e30958979a650d0850fadabb068f6ea17d542a7/test/util/element_type_helpers_test.dart#L87-L106

**Review by greglittlefield-wf — COMMENTED** [review]
Added some comments about potentially reusing existing code, otherwise looks good!

File: lib/src/util.dart
Comment (alanknight-wk) [reply lib/src/util.dart:438]: Oh, yes. I'd seen that, but didn't realize that it actually did what I wanted. Switched to using it, and removed the extension. And now I don't need the test.

**Review by greglittlefield-wf — APPROVED** [review]
Utility-related changes and general approach looks good to me! +1

I don't feel comfortable approving the behavioral changes and handling of bad cases though, so I think review from the INTL team is needed before merge.

File: lib/src/intl_suggestors/intl_migrator.dart
Comment (robbecker-wf) [inline lib/src/intl_suggestors/intl_migrator.dart:16]: Thanks, this is the preferred fixme comment format.

File: lib/src/intl_suggestors/intl_migrator.dart
Comment (robbecker-wf) [inline lib/src/intl_suggestors/intl_migrator.dart:45]: note: this appears to be just a formatting change

Comment (alanknight-wk) [reply lib/src/intl_suggestors/intl_migrator.dart:45]: Yes, my IDE is set to auto-format, and it seemed to change a lot of stuff. Not clear why, this repo isn't set up for dart_dev, so it's just doing a generic dartfmt

**Review by robbecker-wf — APPROVED** [review]
Nice, after using Greg's suggestions this looks good. No further questions.

**Comment by alanknight-wk** [comment]
QA +1
Tested manually, because we don't have a lot of automation on the intl_message_migration.

**Comment by alanknight-wk** [comment]
@Workiva/release-management-p

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**
